Only Show Sync Status to Admin Who Started Gdrive Sync

iethree opened the pull request:
## Description
- use the backend `created-by-id` to only show sync status popup to the admin who connected the google drive folder

Demo:
https://www.loom.com/share/39fa8fa421e54b9fa9474f4065b72c76

iethree committed:
rename and reorganize

# Conflicts:
#	enterprise/frontend/src/metabase-enterprise/google_drive/GdriveConnectionModal.tsx
#	enterprise/frontend/src/metabase-enterprise/google_drive/GdriveSyncStatus.tsx

iethree committed:
resolve conflicts

iethree committed:
only show sync status to admin who created it

sloansparger commented on enterprise/frontend/src/metabase-enterprise/google_drive/GdriveSyncStatus.tsx:
```diff
@@ -21,13 +22,18 @@ export const GdriveSyncStatus = () => {
   const dispatch = useDispatch();
   const isAdmin = useSelector(getUserIsAdmin);
+  const currentUser = useSelector(getCurrentUser);
+  const isCurrentAdmin =
+    isAdmin &&
+    gsheetsSetting?.["created-by-id"] &&
+    currentUser.id === gsheetsSetting?.["created-by-id"];
```
!nit `isSyncInitiator` would be clearer to me

iethree committed:
Add connection info to disconnect modal (#55588)

* rename and reorganize

# Conflicts:
#	enterprise/frontend/src/metabase-enterprise/google_drive/GdriveConnectionModal.tsx
#	enterprise/frontend/src/metabase-enterprise/google_drive/GdriveSyncStatus.tsx

* resolve conflicts

* update disconnect modal

* better spacing
